### ISSUE TITLE
feat[dev]: ✨ implement A11y Re-Improvements to sd-teaser & sd-teaser-media

### ISSUE THREAD

**Issue by yoezlem** [solid-design-system/solid#2063]
## Description
This tasks describes the Figma implementation for: #2048 

This task describes the Storybook implementation for the sd-teaser & sd-teaser-media component as part of the A11y re-improvements outlined in Epic #2048.

A11y improvements for sd-teaser & sd-teaser-media can be found here: [sd-teaser & sd-teaser-media Notes](https://www.notion.so/matuzo/sd-teaser-sd-teaser-media-1baf3f6e1bc081659cfcd1aa962e2d9c).

------------------------------------------

## Comment
**Problem:** Wrapping multiple elements in a link element doesn’t work well.
- If you access the link with the Tab key, a screen-reader reads the entire text contained within the link, the heading, the alt text of the image, the teaser text, and the call to action. Some screen-readers, tell users it’s a link only when they’re done announcing.
- Using the virtual cursor, a screen-reader might announce pieces of text as separate links.
- Nesting links in linked teasers is not possible.
- The text isn’t selectable.
- It’s not clear how to access the link via voice.
- Especially on mobile, users might be unable to access the nested elements separately.
**Solution:** Unfortunately, there is no perfect way to link groups of elements; all solutions come with trade-offs. My favorite solution is to use a pseudo-element that covers the card inside the interactive element. See solution 6 here in this [[comparison of block link solutions](https://codepen.io/matuzo/pen/ZEqbXOM)](https://codepen.io/matuzo/pen/ZEqbXOM) and read my [[comment on GitHub](https://github.com/solid-design-system/solid/issues/1311#issuecomment-2307211838)](https://github.com/solid-design-system/solid/issues/1311#issuecomment-2307211838) where you’ve already addressed the issue..

---

**Problem:** Content is still hard to read with a translucent background. You’re reducing the opacity on hover a little bit. That should also happen on `:focus-within`.

![Image](https://github.com/user-attachments/assets/54cf451e-65dd-47d1-af81-201611d3b445)

---

## Accessibility information
- *Teaser headings should be the first item in the DOM, introducing a new thematic region and separating the following content from the previous region.*
    
    **Yes, but inside the teaser not the entire DOM.**
    
- *Include descriptive alt text for the media slot, ensuring that non-text content is accessible. If the media is purely decorative, place an empty alt text.*
**I would elaborate on that a bit more and explain that images should only be described if the image adds new relevant information to the teaser and only if it's relevant in the context of the teaser. Only because images are not presentational but actually show something that could be described doesn't mean that they need a description. Also mention that the title of the teaser should not be repeated in the description of the image.
This is a good place to write about alt-text best practices in detail.**

## DoR
- [x] Item has been estimated by the team
- [x] Item dependencies have been identified and documented

## DoD
- [ ] Documentation has been created/updated (if applicable)
- [ ] Migration Guide has been created/updated (if applicable)
- [ ] Relevant E2E tests (Features, A11y, Bug fixes) are created/updated
- [ ] Relevant stories (Features, A11y) are created/updated
- [ ] Implementation works successfully on `feature` branch
